src/Models/Scopes/Product/WithProductStockScopeMsi.php

```diff
@@ -14,7 +15,9 @@ class WithProductStockScopeMsi implements Scope
     public function apply(Builder $builder, Model $model)
     {
         // Remove the existing "in_stock" select.
-        $builder->getQuery()->columns = collect($builder->getQuery()->columns)->filter(function ($column) {
+        $builder->getQuery()->columns = collect($builder->getQuery()->columns)->filter(function ($column) use ($builder) {
+            $column = $column instanceof Expression ? $column->getValue($builder->getQuery()->getGrammar()) : $column;
```

Comment:
I got this idea from
https://github.com/laravel/framework/blob/124eeae010f2801d0f4e036c0e47cf9c99cd9dcf/src/Illuminate/Database/Eloquent/Builder.php#L671